ghost/members-api/lib/repositories/member.js

```diff
@@ -841,12 +846,19 @@ module.exports = class MemberRepository {
         }
 
         if (!options.transacting) {
-            return this._Member.transaction((transacting) => {
-                return this.linkSubscription(data, {
+            let executionPromise = null;
+            await this._Member.transaction(async (transacting) => {
+                await this.linkSubscription(data, {
                     ...options,
                     transacting
                 });
+                // Catch updated executionPromise
+                executionPromise = transacting.executionPromise;
             });
+            if (executionPromise) {
+                // Wait for added events
```

Comment:
This is required because we wouldn't wait for SubscriptionCreatedEvents -> unpredictable if those are executed before we 'read' the member back in the response for the same API endpoint = unpredictable responses.